may-2018-hardfork.md

```diff
@@ -0,0 +1,33 @@
+# May 2018 Hardfork Specification
+
+Version 1.0, 2018-03-01
+
+## Summary
+
+When the median time past[1] of the most recent 11 blocks (MTP-11) is greater than or equal to UNIX timestamp 1526400000 Bitcoin Cash will execute a hardfork according to this specification. Starting from the next block these consensus rules changes will take effect:
+
+* Blocksize increase to 32,000,000 bytes
+* Re-enabling of several opcodes
+
+The following are not consensus changes, but are recommended changes for Bitcoin Cash implementations:
+
+* Automatic replay protection for future hardforks
+* Increase OP_RETURN relay size to 223 total bytes
+
+## Blocksize increase
+
+The blocksize hard capacity limit will be increased to 32MB (32000000 bytes).
+
+## OpCodes
+
+Several opcodes will be re-enabled per [may-2018-reenabled-opcodes](may-2018-reenabled-opcodes.md)
+
+## Automatic Replay Protection
+
+When the median time past[1] of the most recent 11 blocks (MTP-11) is greater than or equal to UNIX timestamp 1542300000 Bitcoin Cash full nodes implementing the May 2018 consensus rules SHOULD enforce the following change:
+
+ * Update `forkid`[1] to be equal to 0xFF0001.  ForkIDs beginning with 0xFF will be reserved for future protocol upgrades.
+
+This particular consensus rule MUST NOT be implemented by Bitcoin Cash wallet software.
```

Comment:
I'd like to know more about the intended meaning of this 'MUST NOT' here.

At first glance, it seems inconsistent with the preceding 'SHOULD' in l.27. 

It would be consistent if interpreting 'MUST NOT' as 'DOES NOT HAVE TO', but 'MUST NOT' means absolutely prohibited according to RFC2119.

If wallet software is absolutely prohibited from implementing a future consensus rule, then it will diverge from consensus guaranteed, no?

Reply:
The intent here is that wallets should not implement this rule, only nodes relaying transactions.  That way wallets will stay using the main chain, rather than any future minority fork from Bitcoin Cash.